.github/actions/get-version-info/action.yml

```diff
@@ -0,0 +1,83 @@
+name: 'Get Version Info'
+description: 'Extract version information from git and determine release type'
+outputs:
+  git_commit:
+    description: 'Full Git commit hash'
+    value: ${{ steps.version.outputs.git_commit }}
+  git_version:
+    description: 'Git version string from git describe'
+    value: ${{ steps.version.outputs.git_version }}
+  major:
+    description: 'Major version number'
+    value: ${{ steps.version.outputs.major }}
+  minor:
+    description: 'Minor version number'
+    value: ${{ steps.version.outputs.minor }}
+  build_date:
+    description: 'Build date based on commit timestamp in ISO format'
+    value: ${{ steps.version.outputs.build_date }}
+  release_type:
+    description: 'Determined release type (release, pre-release, or non-release)'
+    value: ${{ steps.version.outputs.release_type }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get version info
+      id: version
+      shell: bash
+      run: |
+        # Get version information
+        git_commit=$(git rev-parse HEAD)
+        git_version=$(git describe --tags --always)
+        
+        # Get commit timestamp (same logic as vcs.time in Go)
+        build_date=$(git show -s --format=%cI $git_commit)
+        
+        echo "==== VERSION INFO ===="
+        echo "Git commit: $git_commit"
+        echo "Git version: $git_version"
+        echo "Build date: $build_date"
+        
+        # Extract major/minor from git version
+        if [[ $git_version =~ ^v?([0-9]+)\.([0-9]+)\..*$ ]]; then
+          major="${BASH_REMATCH[1]}"
+          minor="${BASH_REMATCH[2]}"
+        else
+          major="0"
+          minor="0"
+        fi
```

Comment:
I know that we are trying to get the latest tag and deduce the latest version from it.
Do you think this is the best path for it ?

Reply:
This is the existing behaviour. What other better options you have in mind?
https://github.com/bacalhau-project/bacalhau/blob/254467d4381b02c7fe16d03be3139e246e0ee5be/Makefile#L48